File: CON BDD/EUREKABANK_SOAP_DOTNET_GR06/05. CLIESCRITORIO/Servicio/Sesion.cs
using CLIESCRITORIO.Servicio;

namespace CLIESCRITORIO.Servicio
{
    public class Sesion
    {
        public string? Usuario { get; set; }
        public bool Admin { get; set; }
        public string ClienteAsignado { get; set; } = string.Empty;
        public List<CuentaResumen> Cuentas { get; set; } = new();

        public bool Activa => !string.IsNullOrWhiteSpace(Usuario);

        public bool CuentaPropia(string codigo)
        {
            if (string.IsNullOrWhiteSpace(codigo)) return false;
            return Cuentas.Any(c => codigo == c.CodigoCuenta);
        }
    }
}

File: CON BDD/EUREKABANK_SOAP_JAVA_GR06/04. CLICONSOLA/eurekabank_soap_java_con_gr06/src/main/java/ec/edu/monster/servicio/Sesion.java
package ec.edu.monster.servicio;

import ec.edu.monster.ws.CuentaResumen;
import java.util.ArrayList;
import java.util.List;

/** Estado de la sesión (modelo): usuario, rol y cuentas cargadas. */
public class Sesion {

    private String usuario;
    private boolean admin;
    private String clienteAsignado = "";
    private List<CuentaResumen> cuentas = new ArrayList<>();

    public String getUsuario() { return usuario; }
    public void setUsuario(String v) { this.usuario = v; }

    public boolean isAdmin() { return admin; }
    public void setAdmin(boolean v) { this.admin = v; }

    public String getClienteAsignado() { return clienteAsignado; }
    public void setClienteAsignado(String v) { this.clienteAsignado = v == null ? "" : v; }

    public List<CuentaResumen> getCuentas() { return cuentas; }
    public void setCuentas(List<CuentaResumen> v) {
        this.cuentas = v == null ? new ArrayList<>() : v;
    }

    public boolean activa() { return usuario != null; }

    public boolean cuentaPropia(String codigo) {
        if (codigo == null) return false;
        for (CuentaResumen c : cuentas) {
            if (codigo.equals(c.getCodigoCuenta())) return true;
        }
        return false;
    }
}
